File: Projet_Prograamming/Java/Java_1stYear/NetBeansProjects/g55870.luckynumbers/src/main/java/g55870/luckynumbers/model/Tile.java
package g55870.luckynumbers.model;

public class Tile {

    private final int value;
    private boolean faceUp;

    /**
     * The constructor the class Tile
     *
     * @param value : the value of the tile.
     */
    public Tile(int value) {
        this.value = value;
    }

    /**
     * this method is a getter of the tile
     *
     * @return the value of a tile
     */
    public int getValue() {
        return this.value;
    }

    /**
     * this method is a boolean getter of the tile.
     *
     * @return true if the tile is face up and false otherwise ;
     */
    public boolean isFaceUp() {
        return faceUp;
    }

    /**
     * Flip the tile face up in order to make it visible
     */
    void flipFaceUp() {
        if (!isFaceUp()) {
            this.faceUp = true;
        }
    }
}

File: Projet_Prograamming/C++/ProjetPomme_Balde/metier/Tile.cpp
#include "Tile.h"

Tile::Tile(Color color, bool isApple, bool isPicker ,int value )
    : color(color), isApple(isApple), isPicker(isPicker),value(value) , faceUp(false)
{
}
Tile::~Tile()
{
}
bool Tile::isFaceUp()
{
 return faceUp;
}
void Tile::flipFaceUp()
{
    this->faceUp = true;
}
